integrations/gitlab_v2/main.py

```diff
@@ -0,0 +1,157 @@
+import typing
+from enum import StrEnum
+from typing import Any
+
+from loguru import logger
+from port_ocean.context.event import event
+from port_ocean.context.ocean import ocean
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+
+from client import CREATE_UPDATE_WEBHOOK_EVENTS, DELETE_WEBHOOK_EVENTS, GitlabClient
+from integration import GitlabProjectResourceConfig
+from utils import extract_issue_payload, extract_merge_request_payload
+
+
+class InvalidTokenException(Exception):
+    ...
+
+
+class ResourceKind(StrEnum):
+    GROUP = "group"
+    PROJECT = "project"
+    MERGE_REQUEST = "merge_request"
+    ISSUE = "issue"
+
+
+class WebHookEventType(StrEnum):
+    MERGE_REQUEST = "merge_request"
+    ISSUE = "issue"
+
+
+# Listen to the start event of the integration. Called once when the integration starts.
+@ocean.on_start()
+async def on_start() -> None:
+    logger.info("Starting musah_gitlab integration")
+    tokens = ocean.integration_config["gitlab_access_tokens"]["tokens"]
+
+    # Check if tokens is a list and filter valid tokens (strings only)
+    if isinstance(tokens, list):
+        tokens_are_valid = filter(lambda token: isinstance(token, str), tokens)
+
+        # Ensure all tokens are valid strings
+        if not all(tokens_are_valid):
+            raise InvalidTokenException("Invalid access tokens, ensure all tokens are valid strings")
+    else:
+        raise InvalidTokenException("Invalid access tokens, confirm you passed in a list of tokens")
+
```

Comment:
the validation of tokens should be something that should be handled by GitLab and not us

Reply:
The intent is to support multiple token implementation from different GitLab accounts, allowing for a seamless resync of all resources at once, rather than requiring multiple integrations. However, I can revert to supporting a single token if this approach does not align with the intended requirements. Please let me know if i should stick to just single access token for now.

Reply:
The validation only ensures tokens are provided in below format :
`{"tokens": ["token1", "token2"]}`

Reply:
sure. But ideally, when users provide tokens, we want it to be in this format
{"access-token-1": "list of projects or sub groups it should pull data from, you can also use * so it brings all", "access-token-2": "list of projects or sub groups it should pull data from", "access-token-X": "list of xxx"}

This way, when the user installs the integration, it will fetch data from all the tokens that have been provided

Reply:
https://docs.getport.io/build-your-software-catalog/sync-data-to-catalog/git/gitlab/installation#tokenmapping

You can read more here

Reply:
I see now, thanks for clarifying this.